alf/metrics/metrics.py

```diff
@@ -266,6 +283,114 @@ def _extract_metric_values(self, time_step):
             return [reward[..., i] for i in range(reward.shape[-1])]
 
 
+@gin.configurable
+class AverageDiscountedReturnMetric(AverageEpisodicSumMetric):
+    """Metric for computing the average discounted episodic return.
+    It is calculated according to the following formula:
+
+    .. math::
+        \begin{array}{ll}
+            R &= r_1 + (1 + gamma) r_2 + (1 + gamma + gamma^2) r_3 + \cdots \\
+            &= \frac{1}{L}\sum_{l=1}^L \sum_{k=0}^{l-1} \gamma^k r_l,
+        \end{array}
+    where :math:`\gamma` is the reward discount, and :math:`r_1` denotes the
+    reward due to the first action, which is received at the second time step.
+    :math:`L` equals to the episode length - 1.
+
+
+    Note that if the last step is not due to time limit, the discounted return
+    calculated form the formula above is unbiased. If the last step is due to
+    time limit, it is a biased estimate and its expectation is lower than the
```

Comment:
If the last time step is TimeLimit, then I think a better estimation would be still using the last step's value estimation for the remaining sum.

Reply:
Agree. In this case, the metric is not accurate.

Reply:
For the last several steps, this bias is very severe. If you're going to compare the average metric with the predicted discounted value, then probably the comparison is not informative.